storage/segment_query.go

```diff
@@ -129,7 +129,7 @@ func applyFilter(
 	if err != nil {
 		return nil, err
 	}
-	defer fld.Close()
+	defer toFilter.Close()
```

Comment:
Ah, so it's safe to close `toFilter` after calling `toFilter.Filter` since it creates an iterator that holds references to the underlying data?

Reply:
Actually it's because `queryFields` already holds reference to the underlying data and won't be closed until the query execution is done, so the additional reference `toFilter` holds is not necessary so it can be closed early.